### DISCUSSION TITLE
Javascript API less dependent on UI events

### DISCUSSION THREAD

**Discussion by hovi** [jrief/django-admin-sortable2#307] [General]
Hi,

I checked the sources for the project to look for some kind of API, that allows reordering using different ways.
A few examples:

- A button next to an item in admin table "make this primary", which would automatically set this item in the first position.
- Similarly a button to move the item to the last position.
- Reordering, that reacts to changes in different parts of the formset.

Correct me if I looked wrong, but it's pity, that I can't use the already nice working API to do this functionality, since the functionality is pretty much already there, just mostly hooked on drag & drop events.

The functionality is not complicated and I can just write my own code, but you can consider that next time you refactor your frontend code :-)

**Comment by jrief** [comment]
> A button next to an item in admin table "make this primary", which would automatically set this item in the first position.

you would still have to handle the visual effect of then moving the item to the first/last position. Before implementing something like this, have a look at the API of https://github.com/SortableJS/Sortable – I havn't found any functionality which offers this.

Something you can try is to reorder the DOM yourself and afterwards reorder all `order` fields yourself. Finally you must send them to the server, see [admin-sortablets.ts](https://github.com/jrief/django-admin-sortable2/blob/b8c206a8d4b0f45d7898001b6839c15cfcdd90eb/client/admin-sortable2.ts#L118) on how its done.

> Reordering, that reacts to changes in different parts of the formset.

what do you mean by that?

**Comment by hovi** [comment]
I know I can do it by hand and I can imagine how to do it. I am simply giving you feedback on your ts/js code and it's possible improvements so that it can be more useful.

**Comment by jrief** [comment]
In version 2.1 this feature has been implemented.
